recipes/gtk4/meta.yaml

```
@@ -0,0 +1,136 @@
+{% set name = "gtk" %}
+{% set version = "4.0.3" %}
+{% set version_majmin = '.'.join(version.split('.', 2)[:2]) %}
+
+package:
+  name: gtk4  # note customization
+  version: {{ version }}
+
+source:
+  url: https://download.gnome.org/sources/{{ name }}/{{ version_majmin }}/{{ name }}-{{ version }}.tar.xz
+  sha256: d7c9893725790b50bd9a3bb278856d9d543b44b6b9b951d7b60e7bdecc131890
+  patches:
+    # https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3342
+    - gdkmacosglcontext.c.patch  # [osx]
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('gtk4', max_pin='x') }}
+    # gtk apps need at least a default icon set for fallback
+    - adwaita-icon-theme
+  missing_dso_whitelist:
+    - '*/libX11.so*'  # [linux]
+    - '*/libXext.so*'  # [linux]
+    - /usr/lib/libcups.2.dylib    # [osx]
+
+requirements:
+  build:
+    # restrict to working meson, see https://github.com/conda-forge/gtk3-feedstock/pull/26
+    - meson 0.56.2
+    - ninja
+    - gobject-introspection
+    - pkg-config
+    - {{ compiler('c') }}
+    # sassc needs a C++ compiler
+    - {{ compiler('cxx') }}
+    - {{ cdt('libice-devel') }}  # [linux]
+    - {{ cdt('libsm-devel') }}  # [linux]
+    - {{ cdt('libx11-devel') }}  # [linux]
+    - {{ cdt('libxcomposite-devel') }}  # [linux]
+    - {{ cdt('libxcursor-devel') }}  # [linux]
+    - {{ cdt('libxdamage-devel') }}  # [linux]
+    - {{ cdt('libxext-devel') }}  # [linux]
+    - {{ cdt('libxi-devel') }}  # [linux]
+    - {{ cdt('libxinerama-devel') }}  # [linux]
+    - {{ cdt('libxfixes-devel') }}  # [linux]
+    - {{ cdt('libxrandr-devel') }}  # [linux]
+    - {{ cdt('libxrender-devel') }}  # [linux]
+    - {{ cdt('libxtst-devel') }}  # [linux]
+    - {{ cdt('mesa-libEGL-devel') }}  # [linux]
+    - {{ cdt('mesa-libGL-devel') }}  # [linux]
+    - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
+  host:
+    - cairo
+    - epoxy
+    - freetype
+    - gdk-pixbuf
+    - gettext  # [osx]
+    - glib
+    - libgraphene
+    - harfbuzz
+    - libcups  # [linux]
+    - libiconv  # [win]
+    - pango
+  run:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.12") }}  # [osx and x86_64]
+    - hicolor-icon-theme
+    - pango
+    # The post-links scripts use these
+    - glib-tools
+    - libgraphene
+    - fribidi
+    - freetype
+    - fontconfig
+    - libcups  # [linux]
```

I think we need a pin here since `libcups` doesn't have a run_export (apparently).
```suggestion
    - {{ pin_compatible('libcups') }}  # [linux]
```